mailpoet/lib/Automation/Integrations/MailPoet/Analytics/Endpoints/SubscriberEndpoint.php

```diff
@@ -0,0 +1,52 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Automation\Integrations\MailPoet\Analytics\Endpoints;
+
+use MailPoet\API\REST\Request;
+use MailPoet\API\REST\Response;
+use MailPoet\Automation\Engine\API\Endpoint;
+use MailPoet\Automation\Engine\Exceptions\NotFoundException;
+use MailPoet\Automation\Engine\Storage\AutomationStorage;
+use MailPoet\Automation\Integrations\MailPoet\Analytics\Entities\Query;
+use MailPoet\Automation\Integrations\MailPoet\Analytics\Factories\SubscriberControllerFactory;
+use MailPoet\Validator\Builder;
+
+class SubscriberEndpoint extends Endpoint {
+
+  /** @var AutomationStorage */
+  private $automationStorage;
+
+  /** @var SubscriberControllerFactory */
+  private $subscriberControllerFactory;
+
+  public function __construct(
+    AutomationStorage $automationStorage,
+    SubscriberControllerFactory $subscriberControllerFactory
+  ) {
+    $this->automationStorage = $automationStorage;
+    $this->subscriberControllerFactory = $subscriberControllerFactory;
+  }
+
+  public static function getRequestSchema(): array {
+    return [
+      'id' => Builder::integer()->required(),
+      'query' => Query::getRequestSchema(),
+    ];
+  }
+
+  public function checkPermissions(): bool {
+    return true;
+  }
```

Comment:
Why is this override needed? Would it not make it so unauthenticated users could hit this endpoint and potentially view subscriber data?

Reply:
Thank you for catching that! [075e04f](https://github.com/mailpoet/mailpoet/pull/5038/commits/075e04f20bc18c36cfd14483a44656758611b35f) removes this override. Was a left over for testing purposes.